File: GXpert/GXpert.Web/Modules/Exams/Assignment/AssignmentColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Exams.Columns;

[ColumnsScript("Exams.Assignment")]
[BasedOnRow(typeof(AssignmentRow), CheckNames = true)]
public class AssignmentColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    public string File { get; set; }
    public string ClassTitle { get; set; }
    public string SubjectTitle { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Exams/AssignmentColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AssignmentRow } from "./AssignmentRow";

export interface AssignmentColumns {
    Id: Column<AssignmentRow>;
    Title: Column<AssignmentRow>;
    Description: Column<AssignmentRow>;
    File: Column<AssignmentRow>;
    ClassTitle: Column<AssignmentRow>;
    SubjectTitle: Column<AssignmentRow>;
    InsertDate: Column<AssignmentRow>;
    InsertUserId: Column<AssignmentRow>;
    UpdateDate: Column<AssignmentRow>;
    UpdateUserId: Column<AssignmentRow>;
    IsActive: Column<AssignmentRow>;
}

export class AssignmentColumns extends ColumnsBase<AssignmentRow> {
    static readonly columnsKey = 'Exams.Assignment';
    static readonly Fields = fieldsProxy<AssignmentColumns>();
}